gateway/handlers/forwarding_proxy.go

```diff
@@ -15,6 +16,9 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+// Parse out the service name (group 1) and rest of path (group 2).
+var functionMatcher = regexp.MustCompile("^/?(?:async-)?function/([^/?]+)([^?]*)")
```

Comment:
This needs considering whether `GET` should be allowed for asynchronous functions. Not sure it makes sense to be supported, but if path parameters work for both `function` and `async-function` the user may expect to be able to reference the same path with `async` as without.

Reply:
I think the async route only allows for `POST` right now, but we could extend that. Please @ivanayov check and raise an issue to update if that's the case.

Reply:
Addressed in #829